javascript/app-framework/src/components/DataTableConfig/DataTableConfig.hooks.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { DataFormat, DataOrientation } from '@datashaper/schema'
+import type { DataTable } from '@datashaper/workflow'
+import type { IChoiceGroupOption } from '@fluentui/react'
+import { useCallback } from 'react'
+
+import { useUnobservedProperty } from '../../hooks/index.js'
+
+export function useChangeHandlers(resource: DataTable): {
+	format: DataFormat
+	onChangeFormat: (
+		evt?: React.FormEvent<HTMLElement | HTMLInputElement>,
+		option?: IChoiceGroupOption,
+	) => void
+} {
+	const format = useUnobservedProperty<DataTable, DataFormat>(
+		resource,
+		'format',
```

Comment:
I see, `format` here is a sub-object which will generally be stable - why not use `resource.format`? If it's being mutated like this maybe we should make it observable.